nixos/modules/system/boot/clevis.md

```diff
@@ -0,0 +1,49 @@
+# Clevis {#module-boot-clevis}
+
+[Clevis](https://github.com/latchset/clevis)
+is a framework for automated decryption policy of ressources.
+Clevis allows for secure unattended disk decryption during boot, using decryption policies that must be satisfied for the data to decrypt.
+
+
+## Create a secret {#module-boot-clevis-create-secret}
+
+Secrets have to be created through the clevis command line. 3 types of policies are supported:
+
+1) TPM policies
+
+Secrets are binded against the presence of a TPM2 device, for example:
+```
+echo hi | clevis encrypt tpm2 '{}' > hi.jwe
+```
+2) Tang policies
+
+Secrets are binded against the presence of a Tang server, for example:
+```
+echo hi | clevis encrypt tang '{"url": "http://tang.local"}' > hi.jwe
+```
+
+3) Shamir Secret Sharing
+Secrets are binded against a combination of the two precedent policies, for example:
+```
+echo hi | clevis encrypt sss \
+'{"t": 2, "pins": {"tpm2": {"pcr_ids": "0"}, "tang": {"url": "http://tang.local"}}}' \
+> hi.jwe
+```
+
+For more commplete documentation on how to generate a secret with clevis, see the [clevis documentation](https://github.com/latchset/clevis).
+
+
+## Activate unattended decryption of a ressource at boot {#module-boot-clevis-activate}
+
+In order to activate unattended decryption of a ressource at boot, enable the `clevis` module:
+
+```
+boot.initrd.clevis.enable = true;
+```
+
+Then, specify the device you want to decrypt using a given clevis secret. Clevis will automatically try to decrypt the device at boot and will fallback to interactive unlocking if the decryption policy is not respected.
```

Comment:
"if the decryption policy is not respected" ... what does this mean? That the TPM no longer contains the pin? That the tang server is not accessible, perhaps due to no network devices detected or no network link?

Suggest something other than that word "respect" here. Perhaps "not successful".